### PR TITLE
Tidy up

### PR DESCRIPTION
**Pull request by thomasgoodfellow**
Trivial improvemtns for the example extension: moved to a specific directory as part of adding a minimal test case

### CODE REVIEW

File: examples/cdsl/rv_example/test_subincacc.c
[the remark this answers is not in the document]
Comment (PhilippvK) [reply examples/cdsl/rv_example/test_subincacc.c:1]: See #90 for followup discussions

File: examples/cdsl/rv_example/test_subincacc.c [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+// Compile with the "example" extension specified:
+//   clang test_subincacc.c -c --target=riscv32 -march=rv32ixexample
+// then disassembly shows the new instruction:
+//   llvm-objdump -d test_subincacc.o --disassembler-options=numeric
+
+__attribute__((naked)) void test_subincacc() {
+    asm("li a0, 11");
+    asm("li a1, 5");
+    asm("li a2, 7");
+    asm("cv_subincacc a0, a1, a2");
```
[the remark this answers is not in the document]
Comment (PhilippvK) [reply examples/cdsl/rv_example/test_subincacc.c:14]: See #91 for followup discussions.

File: examples/cdsl/rv_example/test_subincacc.o
[the remark this answers is not in the document]
Comment (PhilippvK) [reply examples/cdsl/rv_example/test_subincacc.o:1]: @thomasgoodfellow After this is pushed, we can merge the PR.

File: examples/corev_demo.py
Comment (thomasgoodfellow) [inline examples/corev_demo.py:1]: This shouldn't be here - was reset locally (--soft, should have been --hard)

**Pull request merged**